frontend/public/components/secrets/create-secret.tsx

```diff
@@ -874,6 +882,15 @@ class GenericSecretForm extends React.Component<GenericSecretFormProps, GenericS
   }
 }
 
+class AlertManagerSecretForm extends GenericSecretForm {
```

Comment:
This is specific to one particular secret in one particular namespace, right? I don't think we should change the generic create secret action. We should instead create an editor just for this config that's separate from these existing secret forms.

Reply:
I'm not changing `GenericSecretForm`, I'm extending it and just overwriting the method(s) I need; thought this was a good practice.

Reply:
> I'm not changing `GenericSecretForm`, I'm extending it and just overwriting the method(s) I need; thought this was a good practice.

I don't think it really makes sense in this case. You aren't sharing any of the fields with the existing secret form, and you wouldn't create a new Alertmanager secret from scratch in your own project (which is what the create-secret form is meant for). You're adding a very specific editor for a single secret in a particular namespace. It's a different use case. The additional props you've had add to change the behavior indicate it's not really the right fit.

It's cleaner to create a separate component that's simply a YAML editor for the Alertmanager secret, then use `k8sPatch` to patch the single value with the updated YAML.

Reply:
Ok, sounds good, I'll look into doing that -thanks :-)

Reply:
fixed, thanks for pointing me in a better coding direction :-)

Reply:
fixed